src/manifolds/ConnectionManifold.jl

```diff
@@ -106,15 +141,25 @@ function christoffel_symbols_second_jacobian(
     M::AbstractManifold,
     p,
     B::AbstractBasis;
-    backend::AbstractDiffBackend=diff_backend(),
+    backend::AbstractDiffBackend=default_differential_backend(),
+    retraction::AbstractRetractionMethod=ManifoldsBase.default_retraction_method(M),
 )
-    n = size(p, 1)
+    d = manifold_dimension(M)
     ∂Γ = reshape(
-        _jacobian(q -> christoffel_symbols_second(M, q, B; backend=backend), p, backend),
-        n,
-        n,
-        n,
-        n,
+        _jacobian(
+            c -> christoffel_symbols_second(
```

Comment:
It looks like a good direction but I'm not quite sure what are the necessary conditions for retraction and basis to make this work.

Reply:
I think it would be a good idea to make an example on a sphere that shows correct/incorrect calculations of its curvature for different bases and retractions.

Reply:
I think I do not follow you here. _Any_ Basis and _any_ retraction give you (locally!) a chart here. Any. It might not work for very large X (projection retraction on the sphere and X of length pi/2), but for small enough X all basis and retraction combinations work.

Reply:
I've elaborated a bit on this here: https://github.com/JuliaManifolds/Manifolds.jl/pull/429#issuecomment-927105712 . The problem is that here we use the basis B at more than one point and there are no requirements on bases that relate these two in any way.

Reply:
To be precise. Let X be a tangent vector (for approximating exp(m,p,X)) lets say on the sphere to have dimension 2. Then in _any_ basis I can write X = b1X1+b2X2 (X1,X2 the basis vectors), so the curve c(t) = t(b1,b2) (in basis parameter space) is the same as c(t) = X(t) (just with get_vector). this only requires _any_ basis. similarly I get a curve on the manifold with _any_ retraction, i.e. d(t) =retr(M,p, get_vetcor(M, p, c(t))). 

Sure if X is too large such that the retraction is not useful anymore (p North Pole, norm of X larger than pi/2, Projectionretraction), this does not work, but locally it always works. But this is the same for _any_ chart you want to get to.